IronFoundry.Container.Shared/ContainerProcess.cs

```diff
@@ -30,6 +30,26 @@ public int WaitForExit()
             return process.ExitCode;
         }
 
-        //int WaitForExit(int milliseconds);
+        /// <summary>
+        /// Wait for the process to exit up to the specified time
+        /// </summary>
+        /// <returns>
+        /// True if the process exited
+        /// </returns>
+        /// <param name="milliseconds">The max time for the process to exit</param>
+        /// <param name="exitCode">Populated with the exit code of the process if it exited</param>
+        public bool WaitForExit(int milliseconds, out int exitCode)
```

Comment:
You were right, the correct name for this would be `TryWaitForExit`.  Sorry I steered you the wrong way.